File: packages/javascript/lib/word/randObject.ts
import type { RandThemedWordOptions, WordDetail } from '../_types/global.js';
import { themedWord } from './wordGenerator.js';

/**
 * Everyday things: what is on a desk, in a bag, around a house.
 *
 * `randWord({ theme: 'object' })` with the theme already chosen; every other
 * option is the same.
 *
 * @example
 * randObject({ language: 'ko', count: 3 }); // ['물병', '연필', '우산']
 * randObject({ language: 'en', count: 3 }); // ['Bottle', 'Pencil', 'Umbrella']
 */
export function randObject(options?: RandThemedWordOptions & { output?: 'value' }): string[];
/**
 * Object words with the language and theme each one came from.
 *
 * @example
 * randObject({ language: 'ko', output: 'detail' });
 * // [{ word: '물병', language: 'ko', theme: 'object' }]
 */
export function randObject(options: RandThemedWordOptions & { output: 'detail' }): WordDetail[];
export function randObject(options: RandThemedWordOptions = {}): string[] | WordDetail[] {
	return themedWord('object', options);
}

File: packages/dart/lib/src/word/rand_object.dart
import 'dart:math';

import 'package:randino/src/types.dart';
import 'package:randino/src/word/rand_word.dart';

/// Everyday things: what is on a desk, in a bag, around a house.
///
/// [randWord] with the theme already chosen; every other parameter is the same.
/// For the detail form, pass [WordTheme.object] to `randWordDetails` — Dart has
/// no overloads, and twenty-four more functions for it would be twenty-four too many.
///
/// ```dart
/// randObject(language: WordLanguage.ko, count: 3); // [물병, 연필, 우산]
/// randObject(language: WordLanguage.en, count: 3); // [Bottle, Pencil, Umbrella]
/// ```
List<String> randObject({
  WordLanguage? language,
  int count = 1,
  RandRealism realism = RandRealism.real,
  RandVocabulary vocabulary = RandVocabulary.full,
  int? minLength,
  int? maxLength,
  String? startsWith,
  bool unique = false,

  /// Where the randomness comes from: `Random.secure()` for a value nobody may
  /// predict, `Random(42)` for one that has to come out the same every run.
  Random? random,
}) => randWord(
  language: language,
  theme: WordTheme.object,
  count: count,
  realism: realism,
  vocabulary: vocabulary,
  minLength: minLength,
  maxLength: maxLength,
  startsWith: startsWith,
  unique: unique,
  random: random,
);
